### PR TITLE
[data] assert metadata size_bytes is int

### PR DESCRIPTION
**Pull request by Zandew**
## Why are these changes needed?

Follow up to #40819 

Asserts that `BlockMetadata.size_bytes` is an int, and updates the remaining references that pass an incorrect type.

## Checks

- [ ] I've signed off every commit(by using the -s flag, i.e., `git commit -s`) in this PR.
- [ ] I've run `scripts/format.sh` to lint the changes in this PR.
- [ ] I've included any doc changes needed for https://docs.ray.io/en/master/.
    - [ ] I've added any new APIs to the API Reference. For example, if I added a 
           method in Tune, I've added it in `doc/source/tune/api/` under the 
           corresponding `.rst` file.
- [ ] I've made sure the tests are passing. Note that there might be a few flaky tests, see the recent failures at https://flakey-tests.ray.io/
- Testing Strategy
   - [ ] Unit tests
   - [ ] Release tests
   - [ ] This PR is not tested :(

### CODE REVIEW

File: python/ray/data/tests/test_stats.py [hunk 1]
```diff
@@ -1208,7 +1208,7 @@ def sleep_three(x):
 
 
 def test_stats_actor_iter_metrics():
-    ds = ray.data.range(1e6).map_batches(lambda x: x)
```
Comment (Zandew) [inline python/ray/data/tests/test_stats.py:1211]: Should we check in `RangeDatasource` that `n` is an int? Using scientific notation results in a float

Comment (bveeramani) [reply python/ray/data/tests/test_stats.py:1211]: What happens if we pass a float to `range`? Do we get the `StatsActor` error?

Comment (Zandew) [reply python/ray/data/tests/test_stats.py:1211]: `StatsActor` works with float, it's just for the `BlockMetadata.size_bytes` typing

Comment (bveeramani) [reply python/ray/data/tests/test_stats.py:1211]: I see.

> Should we check in RangeDatasource that n is an int? Using scientific notation results in a float

What do you have in mind? Like adding an `assert isinstance(n, int)` or raise a `ValueError`?

Comment (Zandew) [reply python/ray/data/tests/test_stats.py:1211]: Yeah I think raising a `ValueError` could work here. Maybe it's too harsh? We could also just try casting here. What do you think?

Comment (raulchen) [reply python/ray/data/tests/test_stats.py:1211]: maybe just convert the input n to int. scientific notation is convenient to use. raising errors seems to cause unnecessary inconvenience

**Review by raulchen — APPROVED** [review]

File: python/ray/data/tests/test_stats.py (hunk 1 above)
Comment (bveeramani) [reply python/ray/data/tests/test_stats.py:1211]: Yeah, +1 to casting input to `int`. If there's some loss in data (e.g, going from 10.6 to 10), we could raise a warning.

While defensive programming is good, I think it's also a good practice to define errors out of existence (Chapter 10 of [Philosophy of Software Design](https://milkov.tech/assets/psd.pdf))

**Review by bveeramani — APPROVED** [review]

File: python/ray/data/tests/test_stats.py [hunk 2]
```diff
@@ -1208,7 +1208,7 @@ def sleep_three(x):
...
+    ds = ray.data.range(100).map_batches(lambda x: x)
```
Comment (raulchen) [inline python/ray/data/tests/test_stats.py:1211]: let's revert this change back then?
